src/utils/cost.ts

```diff
@@ -2,15 +2,48 @@ import { ModelInfo } from "../shared/api"
 
 function calculateApiCostInternal(
 	modelInfo: ModelInfo,
-	inputTokens: number,
+	inputTokens: number, // Note: For OpenAI-style, this is non-cached tokens. For Anthropic-style, this is total input tokens.
 	outputTokens: number,
 	cacheCreationInputTokens: number,
 	cacheReadInputTokens: number,
+	totalInputTokensForPricing?: number, // The *total* input tokens, used for tiered pricing lookup
 ): number {
+	// Determine effective input price
+	let effectiveInputPrice = modelInfo.inputPrice || 0
+	if (modelInfo.inputPriceTiers && modelInfo.inputPriceTiers.length > 0 && totalInputTokensForPricing !== undefined) {
+		// Ensure tiers are sorted by tokenLimit ascending before finding
+		const sortedInputTiers = [...modelInfo.inputPriceTiers].sort((a, b) => a.tokenLimit - b.tokenLimit)
+		// Find the first tier where the total input tokens are less than or equal to the limit
+		const tier = sortedInputTiers.find((t) => totalInputTokensForPricing! <= t.tokenLimit)
+		if (tier) {
+			effectiveInputPrice = tier.price
+		} else {
+			// Should ideally not happen if Infinity is used for the last tier, but fallback just in case
+			effectiveInputPrice = sortedInputTiers[sortedInputTiers.length - 1]?.price || 0
+		}
+	}
+
+	// Determine effective output price (based on total *input* tokens for pricing)
+	let effectiveOutputPrice = modelInfo.outputPrice || 0
+	if (modelInfo.outputPriceTiers && modelInfo.outputPriceTiers.length > 0 && totalInputTokensForPricing !== undefined) {
```

Comment:
The logic for finding the correct price tier is repeated for both input and output prices so make it out into a funciton

Reply:
I think the complexity here is small enough that a refactor isn't needed.